### PR TITLE
platform automation changes for Azure

### PR DESCRIPTION
**Pull request by jtarchie**
These PRs are list of changes that were required to get Platform Automation's CI to work.

### CODE REVIEW

File: azure/dns.tf [hunk 1]
```diff
@@ -3,55 +3,55 @@ data "azurerm_dns_zone" "hosted" {
 }
 
 resource "azurerm_dns_a_record" "opsmanager" {
-  name                = "opsmanager"
+  name                = "opsmanager.${var.environment_name}"
```
Comment (voor) [inline azure/dns.tf]: I think I see where the confusion was coming from with the absence of these.  My hosted zone includes the `environment_name` in it already, so by putting this in here I end up with duplications.

Comment (jtarchie) [reply azure/dns.tf]: I am was aiming for the consistency.
The AWS and GCP both have the `opsmanager` + `env_name`.

Comment (voor) [reply azure/dns.tf]: Screenshot from 2020-01-16 16-32-02

Comment (jtarchie) [reply azure/dns.tf]: Is it? Please bring up the discussion in slack. I'm willing to bet this is one of those, _it depends on the customer_, scenarios.

**Comment by voor** [comment]
I can create an override to make the DNS changes.

For idempotency, I needed to add this stuff (see note at top of [here](https://www.terraform.io/docs/providers/azurerm/r/subnet_network_security_group_association.html)):

```diff
@@ -11,6 +11,8 @@ resource "azurerm_subnet" "management" {
   resource_group_name  = azurerm_resource_group.platform.name
   virtual_network_name = azurerm_virtual_network.platform.name
   address_prefix       = local.management_subnet_cidr
+
+  network_security_group_id = azurerm_network_security_group.ops-manager.id
 }
 
 resource "azurerm_subnet_network_security_group_association" "ops-manager" {
@@ -31,6 +33,7 @@ resource "azurerm_subnet" "pas" {
   resource_group_name  = azurerm_resource_group.platform.name
   virtual_network_name = azurerm_virtual_network.platform.name
   address_prefix       = local.pas_subnet_cidr
+  network_security_group_id = azurerm_network_security_group.platform-vms.id
 }
 
 resource "azurerm_subnet_network_security_group_association" "pas" {
@@ -44,6 +47,7 @@ resource "azurerm_subnet" "services" {
   resource_group_name  = azurerm_resource_group.platform.name
   virtual_network_name = azurerm_virtual_network.platform.name
   address_prefix       = local.services_subnet_cidr
+  network_security_group_id = azurerm_network_security_group.platform-vms.id
 }
 
 resource "azurerm_subnet_network_security_group_association" "services" {

```

Also there's a few changes when you run `terraform fmt`

**Comment by voor** [comment]
Again for idempotency, we need to add some tag ignoring stuff to the BOSH storage account, or else the BOSH added tags will get wiped out by terraform refreshes:

```
     ~ tags                              = {
          + "account_for" = "bosh"
          + "environment" = "test"
          - "type"        = "stemcell" -> null
          - "user-agent"  = "bosh" -> null
        }
```

This achieved by adding this:

```diff
diff --git a/azure/storage.tf b/azure/storage.tf
index b63d67c..396b4d7 100644
--- a/azure/storage.tf
+++ b/azure/storage.tf
@@ -81,6 +81,13 @@ resource "azurerm_storage_account" "bosh" {
     environment = var.environment_name
     account_for = "bosh"
   }
+
+  lifecycle {
+    create_before_destroy = true
+    ignore_changes = [
+      tags
+    ]
+  }
 }
 
 resource "azurerm_storage_container" "bosh_storage_container" {
```

**Comment by jtarchie** [comment]
That is a fair thing to add. @genevieve, maybe we can do the acceptance test, after a director and PAS/PKS deploymnet, rerun terraform, make sure nothing changes.

**Comment by genevieve** [comment]
> I can create an override to make the DNS changes.
> 
> For idempotency, I needed to add this stuff (see note at top of [here](https://www.terraform.io/docs/providers/azurerm/r/subnet_network_security_group_association.html)):
> 
> ```diff
> @@ -11,6 +11,8 @@ resource "azurerm_subnet" "management" {
>    resource_group_name  = azurerm_resource_group.platform.name
>    virtual_network_name = azurerm_virtual_network.platform.name
>    address_prefix       = local.management_subnet_cidr
> +
> +  network_security_group_id = azurerm_network_security_group.ops-manager.id
>  }
>  
>  resource "azurerm_subnet_network_security_group_association" "ops-manager" {
> @@ -31,6 +33,7 @@ resource "azurerm_subnet" "pas" {
>    resource_group_name  = azurerm_resource_group.platform.name
>    virtual_network_name = azurerm_virtual_network.platform.name
>    address_prefix       = local.pas_subnet_cidr
> +  network_security_group_id = azurerm_network_security_group.platform-vms.id
>  }
>  
>  resource "azurerm_subnet_network_security_group_association" "pas" {
> @@ -44,6 +47,7 @@ resource "azurerm_subnet" "services" {
>    resource_group_name  = azurerm_resource_group.platform.name
>    virtual_network_name = azurerm_virtual_network.platform.name
>    address_prefix       = local.services_subnet_cidr
> +  network_security_group_id = azurerm_network_security_group.platform-vms.id
>  }
>  
>  resource "azurerm_subnet_network_security_group_association" "services" {
> ```
> 
> Also there's a few changes when you run `terraform fmt`

Hey @voor 

We originally had these propertiesbut then received this message:
```
Warning: "network_security_group_id": [DEPRECATED] Use the `azurerm_subnet_network_security_group_association` resource instead.
```
Here are the versions I'm using:
```
 ‣ terraform -v
Terraform v0.12.19
+ provider.azurerm v1.33.0
+ provider.random v2.2.1
+ provider.tls v2.1.1
```

Perhaps I'm missing something?

**Comment by voor** [comment]
@genevieve you're not missing anything, it's just confusing message since they need to warn folks about the depreciation while still requiring it.

**Comment by genevieve** [comment]
@voor Awesome, thank you. Updated the template and added a note about the deprecation notice so we don't accidentally delete again the next time someone sees the `Warning`..

**Pull request merged**